### DISCUSSION TITLE
Update MSSQL jdbc driver

### DISCUSSION THREAD

**Discussion by lampi87** [nextgenhealthcare/connect#5590] [Q&A] [answered]
The currently used mssql jdbc driver has version 8.4.1 and is related to jre8
Is there any possibility to update this jdbc driver?

At the moment I'm only able to use openjdk8 with mssql and integrated authentication/Kerberos on Ubuntu.
If I change the used java version `update-alternatives --config java` to openjdk11 or 17 I get a lot of exceptions like
```
mcservice[896969]: ERROR 2022-12-11 10:30:52.996 [Main Server Thread] com.mirth.connect.server.Mirth: Error establishing connection to database, retrying startup in 10000 milliseconds
mcservice[896969]: com.zaxxer.hikari.pool.HikariPool$PoolInitializationException: Failed to initialize pool: Integrated authentication failed. ClientConnectionId:d54c51dd-b19b-4c57-9f17-a0e40981bf52
...
mcservice[896969]: Caused by: com.microsoft.sqlserver.jdbc.SQLServerException: Integrated authentication failed. ClientConnectionId:d54c51dd-b19b-4c57-9f17-a0e40981bf52
...
mcservice[896969]: Caused by: org.ietf.jgss.GSSException: No valid credentials provided (Mechanism level: KDC has no support for encryption type (14))
...
mcservice[896969]: Caused by: sun.security.krb5.KrbException: KDC has no support for encryption type (14)
...
mcservice[896969]: Caused by: sun.security.krb5.Asn1Exception: Identifier doesn't match expected value (906)
```

Best regards
Alexander

**Comment by ab-20-m** [comment, marked as answer]
You should be able to stop the Mirth service, remove the old JAR and replace with a newer version of the JDBC driver.

I doubt a newer driver is going to resolve the authentication issues you are seeing after swapping to Java 11/17. What specific version of Java 8 is working versus Java 11/17? That is: Java 8u341 works, Java 11.0.17 fails, all Java 17.0.X fail. There was a change introduced in Java 8u351/11.0.17 releases that [Disabled 3DES and RC4 in Kerberos](https://www.java.com/en/configure_crypto.html#Disable3DESandRC4). Java 17 always had those algorithms disabled/marked weak by default for Kerberos.

**Reply by lampi87** [reply]
Thank you very much. Aes256 was not enabled for DB service user

**Comment by wi4seo** [comment]
Hi,

FREE Registration for Mirth Unconference 2023 is Now Opened!

To know more, please visit the event page and register yourself: https://wi4.org/mirth-connect-unconference-feb-2023/

Stay Tuned for More Updates!​

​Thanks
